pkg/kn/flags/podspec_helper.go

```diff
@@ -0,0 +1,611 @@
+
+// UpdateContainerPort updates container with a given name:port
+func UpdateContainerPort(spec *corev1.PodSpec, port string) error {
+	container, err := containerOfPodSpec(spec)
+	if err != nil {
+		return err
+	}
+
+	var containerPort int64
+	var name string
+
+	elements := strings.SplitN(port, ":", 2)
+	if len(elements) == 2 {
+		name = elements[0]
+		containerPort, err = strconv.ParseInt(elements[1], 10, 32)
+		if err != nil {
+			return fmt.Errorf(PortFormatErr, port)
+		}
+	} else {
+		name = ""
+		containerPort, err = strconv.ParseInt(elements[0], 10, 32)
+		if err != nil {
+			return fmt.Errorf(PortFormatErr, port)
+		}
+	}
+
+	container.Ports = []corev1.ContainerPort{{
+		ContainerPort: int32(containerPort),
+		Name:          name,
+	}}
+	return nil
+}
+
+// UpdateUser updates container with a given user id
+func UpdateUser(spec *corev1.PodSpec, user int64) error {
+	container, err := containerOfPodSpec(spec)
+	if err != nil {
+		return err
+	}
+	container.SecurityContext = &corev1.SecurityContext{
+		RunAsUser: &user,
+	}
+	return nil
+}
+
+// UpdateResources updates container resources for given revision spec
+func UpdateResources(spec *corev1.PodSpec, resources corev1.ResourceRequirements, requestsToRemove, limitsToRemove []string) error {
+	container, err := containerOfPodSpec(spec)
+	if err != nil {
+		return err
+	}
+
+	if container.Resources.Requests == nil {
+		container.Resources.Requests = corev1.ResourceList{}
+	}
+
+	for k, v := range resources.Requests {
+		container.Resources.Requests[k] = v
+	}
+
+	for _, reqToRemove := range requestsToRemove {
+		delete(container.Resources.Requests, corev1.ResourceName(reqToRemove))
+	}
+
+	if container.Resources.Limits == nil {
+		container.Resources.Limits = corev1.ResourceList{}
+	}
+
+	for k, v := range resources.Limits {
+		container.Resources.Limits[k] = v
+	}
+
+	for _, limToRemove := range limitsToRemove {
+		delete(container.Resources.Limits, corev1.ResourceName(limToRemove))
+	}
+
+	return nil
+}
+
+// UpdateResourcesDeprecated updates resources as requested
+func UpdateResourcesDeprecated(spec *corev1.PodSpec, requestsResourceList corev1.ResourceList, limitsResourceList corev1.ResourceList) error {
+	container, err := containerOfPodSpec(spec)
+	if err != nil {
+		return err
+	}
+	if container.Resources.Requests == nil {
+		container.Resources.Requests = corev1.ResourceList{}
+	}
+
+	for k, v := range requestsResourceList {
+		container.Resources.Requests[k] = v
+	}
+
+	if container.Resources.Limits == nil {
+		container.Resources.Limits = corev1.ResourceList{}
+	}
+
+	for k, v := range limitsResourceList {
+		container.Resources.Limits[k] = v
+	}
+
+	return nil
+}
+
+// UpdateServiceAccountName updates the service account name used for the corresponding knative service
+func UpdateServiceAccountName(spec *corev1.PodSpec, serviceAccountName string) error {
+	serviceAccountName = strings.TrimSpace(serviceAccountName)
+	spec.ServiceAccountName = serviceAccountName
+	return nil
```

Comment:
Could we get rid of returning error here? It doesn't seem to carry any content.

Reply:
sure. Thank you.